02_bubble/pfannerj/main_test.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	mainout = &buf
+
+	main()
+
+	expected := `Bubble sort starting...
+[1 2 3 5]
+Bubble sort completed...
+Insertion sort starting...
+[1 2 3 5]
+Insertion sort completed...
+`
+	actual := buf.String()
+	if expected != actual {
+		t.Errorf(expected, actual, "Unexpected output in main()")
+	}
+}
+
+func TestBubbleOutputFour(t *testing.T) {
+
+	slice := bubble([]int{3, 2, 1, 5})
+
+	expected := [4]int{1, 2, 3, 5}
+	actual := [4]int{}
+	for i, v := range slice {
+		actual[i] = v
+	}
+	if expected != actual {
+		t.Errorf("Unexpected output in bubble()")
+	}
+}
+
+func TestBubbleOutputEight(t *testing.T) {
+
+	slice := bubble([]int{6, 3, 8, 7, 5, 4, 1, 2})
+
+	expected := [8]int{1, 2, 3, 4, 5, 6, 7, 8}
+	actual := [8]int{}
+	for i, v := range slice {
+		actual[i] = v
+	}
+	if expected != actual {
+		t.Errorf("Unexpected output in bubble()")
+	}
+}
+
+func TestBubbleOutputRepeatedNbr(t *testing.T) {
+
+	slice := bubble([]int{6, 3, 1, 7, 5, 4, 1, 2})
+
+	expected := [8]int{1, 1, 2, 3, 4, 5, 6, 7}
+	actual := [8]int{}
+	for i, v := range slice {
```

Comment:
In go, if index(i) is not used, it can be written as `for _, test := range tests `.  The underscore can be replaced with index i if it's going to be used. Actual should be `acutal :=bubble(testInput)` etc.

Reply:
Will bear this in mind while I'm changing the tests to be table driven..